### ISSUE TITLE
NFT fset & iset files not loading consistently, stuck at "Loading, please wait..." screen

### ISSUE THREAD

**Issue by object404** [AR-js-org/AR.js#196]
**Do you want to request a *feature* or report a *bug*?**
Reporting a bug

**What is the current behavior?**
Half the time the AR.js experience hangs and it might be caused by the NFT fset, fset3, and iset files possibly not loading.

Console logs are stuck at "[info] Labeling mode set to 1" and the experience is stuck at the "Loading, please wait..." screen.

After reloading the browser a few times, it gets past it and continues to the AR experience, but the inconsistency of getting past the loading screen is very worrying.

**If the current behavior is a bug, please provide the steps to reproduce.**
I used the NFT Video pinball example at:

https://github.com/AR-js-org/AR.js/tree/master/aframe/examples/image-tracking/nft-video

and the T-Rex example at:
https://github.com/AR-js-org/AR.js/tree/master/aframe/examples/image-tracking/nft

**Please mention other relevant information such as the browser version, Operating System and Device Name**
iOS Safari, iPhone 7
Chrome Android 9

**What is the expected behavior?**

That the experience gets past the loading screen

**If this is a feature request, what is motivation or use case for changing the behavior?**

The inconsistency of getting past the loading screen or not is a dealbreaker for using the AR.js library and recommending it to other devs :-(

**Comment by object404** [comment]
I uploaded the pinball example to my server and it gets stuck on Loading half the time:

https://object404.com/arpinball/

**Comment by object404** [comment]
Here's a screenshot of the Chrome devtools logs:

https://object404.com/arpinball/arpinballstuck.png

**Comment by kalwalt** [comment]
Hi @object404 which type of server are you using?

**Comment by object404** [comment]
I use no special backend. It's just a typical server.

**Comment by kalwalt** [comment]
I tested in my desktop machine and he is loading all the times, I know that loading is slow. @ThostenBux and @Carnaux tried to develop in jsartoolkit5 a system to speed up the process but it is quite experimental, see this [issue](https://github.com/kalwalt/jsartoolkit5/issues/34)

**Comment by Carnaux** [comment]
For testing the compression system in jsartoolkit5:

The branch of jsartoolkit5: https://github.com/Carnaux/jsartoolkit5/tree/compression

The branch of the NFT-Marker-Creator: https://github.com/Carnaux/NFT-Marker-Creator/tree/compression-master

Be aware that this jsartoolkit5 branch is older than the current ones.

**Comment by kalwalt** [comment]
Daniel would be nice continue the dev of this... :slightly_smiling_face:

**Comment by object404** [comment]
Okay I'll try making a marker this one: https://github.com/Carnaux/NFT-Marker-Creator/tree/compression-master

I'll update you guys!

**Comment by object404** [comment]
Okay, I tried downloading the above tree as ZIP, did npm install,

then when I type npm app.js -i markerImage.png, it didn't run unlike in the master branch and this is what the console resulted in:

--------------

Usage: npm <command>

where <command> is one of:
    access, adduser, audit, bin, bugs, c, cache, ci, cit,
    clean-install, clean-install-test, completion, config,
    create, ddp, dedupe, deprecate, dist-tag, docs, doctor,
    edit, explore, fund, get, help, help-search, hook, i, init,
    install, install-ci-test, install-test, it, link, list, ln,
    login, logout, ls, org, outdated, owner, pack, ping, prefix,
    profile, prune, publish, rb, rebuild, repo, restart, root,
    run, run-script, s, se, search, set, shrinkwrap, star,
    stars, start, stop, t, team, test, token, tst, un,
    uninstall, unpublish, unstar, up, update, v, version, view,
    whoami

npm <command> -h  quick help on <command>
npm -l            display full usage info
npm help <term>   search for help on <term>
npm help npm      involved overview

Specify configs in the ini-formatted file:
    C:\Users\pcuser\.npmrc
or on the command line via: npm <command> --key value
Config info can be viewed via: npm help config

npm@6.14.5 C:\Program Files\nodejs\node_modules\npm

**Comment by kalwalt** [comment]
@object404 i think you should run the command with node not npm:
`node app.js -i markerImage.png`

**Comment by object404** [comment]
Ugh I'm an idiot! That worked, thanks! (sorry lack sleep)

**Comment by jwebking** [comment]
I was having the same issue with the loading screen. Adding these details in case they help anyone else.

**Generate your image here:**
https://carnaux.github.io/NFT-Marker-Creator/

**Make sure these two scripts are in your index.html file.** 
<script src="https://cdn.jsdelivr.net/gh/aframevr/aframe@1c2407b26c61958baa93967b5412487cd94b290b/dist/aframe-master.min.js"></script><br><br>
<script src="https://raw.githack.com/AR-js-org/AR.js/master/aframe/build/aframe-ar-nft.js"></script><br><br>

**Models and CORS:**
For your nft url or entity-model url, you can use: https://arjs-cors-proxy.herokuapp.com/ {path to your raw github code or file}

**Comment by everythingability** [comment]
Yep... had all those... but thanks.

**Comment by kalwalt** [comment]
Maybe this [stackoverflow article](https://stackoverflow.com/questions/61262063/image-tracking-and-location-based-ar-with-a-frame-and-ar-js-3-having-a-problem-w/61349001?noredirect=1#comment108867954_61349001) can be useful?

**Comment by mchangxe** [comment]
any update to this???? it takes me 10min + to finish loading and most of the times the page just seems to be stuck. Any ideas?

**Comment by PedroNan** [comment]
I have a similar problem, it works with the trex image in the example, but when I change this image to a custom one it doesn't work and the app gets stuck on the "loading, please wait" message and never moves from here. Any ideas how to fix it?

**Comment by Ebeleh** [comment]
> I have a similar problem, it works with the trex image in the example, but when I change this image to a custom one it doesn't work and the app gets stuck on the "loading, please wait" message and never moves from here. Any ideas how to fix it?

I have this problem too . do u know how to fix it?

**Comment by Seeingred0** [comment]
Yep, seems like a big problem for me alos, even the main example on codepen takes forever to load - https://codepen.io/nicolocarpignoli/pen/vYOeYKd?editors=1010

**Comment by pipboy9000** [comment]
I'm getting the same problem, any news on this?

**Comment by kashishs1993** [comment]
Same problem. Any update?

**Comment by alejandrogamezg** [comment]
I have the same problem. Has anyone been able to solve?

**Comment by dangilbert1** [comment]
Trex example also ( codepen ) or custom Image.  Not loading any Image descriptors , stuck on loading constantly.  Has this not got any further forward now  ?.  Oct. 2024

**Comment by ThorstenBux** [comment]
Yeah, if you look through the console logs in the codepen example: https://codepen.io/nicolocarpignoli/pen/vYOeYKd?editors=1000
You will see that there are a couple of broken urls and once you fix them you will find this
```
Refused to create a worker from 'blob:https://cdpn.io/ae1845db-4154-4632-9d08-74dbbab5949d' because it violates the following Content Security Policy directive: "script-src 'none'". Note that 'worker-src' was not explicitly set, so 'script-src' is used as a fallback.
```
and
```
ae1845db-4154-4632-9d08-74dbbab5949d:1 [Report Only] Refused to compile or instantiate WebAssembly module because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'none'".
```
Which prevents the example from running. The author would need to update the example @nicolocarpignoli should be the one I assume?

**Comment by dangilbert1** [comment]
Thanks . So it would seem from digging that AR.js is unmaintained now then. R.I.P

**Comment by nickw1** [comment]
@dangilbert1 we need more maintainers, particularly on the NFT and marker-based side. Might also be worth talking to @kalwalt.

In terms of the location-based side, this also seen little updates recently but I am proposing re-starting it as a separate project within AR.js, with an entirely separate codebase to NFT/marker-based to help with debugging/updates. Again dependent on my time availability.
@ThorstenBux maybe this may be of interest.
